File: AnnotationProcessorDemo/app/src/main/java/com/apkfuns/annotationprocessordemo/MainActivity.java
package com.apkfuns.annotationprocessordemo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.apkfuns.annotationprocessordemo.utils.Utils;
import com.apkfuns.annotations.BindClass;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * https://github.com/JakeWharton/butterknife
 */
@BindClass(name = "TestActivity")
public class MainActivity extends AppCompatActivity {

    @BindView(R.id.title)
    TextView title;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        title.setText("123");
    }

    @OnClick(R.id.title)
    public void onTextClick(View view) {
        Utils.showToast(this, "Hello World");
    }


}

File: HenCoderSample/toucheventsample/src/main/java/com/apkfuns/toucheventsample/MainActivity.kt
package com.apkfuns.toucheventsample

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.MotionEvent
import android.view.TouchDelegate
import android.view.View
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    companion object {
        @JvmStatic
        val TAG:String = "MainActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<TextView>(R.id.text).setOnTouchListener(View.OnTouchListener { view, motionEvent ->
            Log.d(TAG, "View setOnTouchListener")
            true
        })

    }

    override fun dispatchTouchEvent(ev: MotionEvent?): Boolean {
        val bool = super.dispatchTouchEvent(ev)
//        Log.d(TAG, "dispatchTouchEvent:$bool")
        return true
    }

    override fun onTouchEvent(event: MotionEvent?): Boolean {
        val bool = super.onTouchEvent(event)
        Log.d(TAG, "onTouchEvent:$bool")
        return bool
    }
}
